unidist/core/backends/mpi/core/communication.py

```diff
@@ -850,3 +969,36 @@ def recv_serialized_data(comm, source_rank):
     """
     s_buffer = mpi_recv_buffer(comm, source_rank)
     return SimpleDataSerializer().deserialize_pickle(s_buffer)
+
+
+def send_reserve_operation(comm, data_id, data_size):
+    """
+    Reserve shared memory for data_id
+
+    Parameters
+    ----------
+    data_id : unidist.core.backends.common.data_id.DataID
+    data_size : int
+        Length of required range in shared memory
+
+    Returns
+    -------
+    dict
+        Reservation info about the allocated range in shared memory
+    """
+    operation_type = common.Operation.RESERVE_SHARED_MEMORY
+    mpi_state = MPIState.get_instance()
+
+    operation_data = {
+        "id": data_id,
+        "size": data_size,
+    }
+    # We use a blocking send here because we have to wait for
+    # completion of the communication, which is necessary for the pipeline to continue.
+    send_simple_operation(
+        comm,
+        operation_type,
+        operation_data,
+        mpi_state.get_monitor_by_worker_rank(),
```

Comment:
Do we reserve a place for data on all hosts? Why?

Reply:
Some data stored on one host may not be stored on another host. Therefore, to avoid unnecessary memory usage, we reserve space only on one host, and on others - separately as needed.

Reply:
I see, thanks!